tests/helpers/shared-test-setups/typescript.js

```diff
@@ -0,0 +1,38 @@
+const options = {
+  addonLocation: undefined,
+  projectRoot: 'tmp/ember-container-query-typescript',
+  testAppLocation: undefined,
+  testAppName: undefined,
+};
+
+const augmentedOptions = {
+  locations: {
+    addon: 'ember-container-query',
+    testApp: 'test-app',
+  },
+  packageManager: {
+    isNpm: false,
+    isPnpm: false,
+    isYarn: true,
+  },
+  packages: {
+    addon: {
+      dependencies: new Map([
+        ['ember-cli-babel', '^7.26.11'],
+        ['ember-cli-htmlbars', '^6.1.1'],
+        ['typescript', '^4.9.4'],
+      ]),
```

Comment:
I may update the list of dependencies as I write tests for the remaining steps. These 3 form the minimum set that I need to consider for now.